### ISSUE TITLE
duplicate column name warnings in SQLite log

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#8990]
Since a couple of days (don't know exactly since when) I'm getting the following warnings on startup:

```
2020-09-11 22:10:03.397 18612-18612/cgeo.geocaching E/SQLiteLog: (1) duplicate column name: image_title_prefix
2020-09-11 22:10:03.397 18612-18612/cgeo.geocaching E/SQLiteLog: (1) duplicate column name: image_scale
2020-09-11 22:10:03.397 18612-18612/cgeo.geocaching E/SQLiteLog: (1) duplicate column name: favorite
2020-09-11 22:10:03.398 18612-18612/cgeo.geocaching E/SQLiteLog: (1) duplicate column name: rating
2020-09-11 22:10:03.398 18612-18612/cgeo.geocaching E/SQLiteLog: (1) duplicate column name: password
2020-09-11 22:10:03.398 18612-18612/cgeo.geocaching E/SQLiteLog: (1) duplicate column name: tweet
```

@eddiemuc 
This is probably related to the offline logging & image changes. Can you have a look at it?

**Comment by eddiemuc** [comment]
I will take a look. This is definitely related

**Comment by eddiemuc** [comment]
Hi @moving-bits , these messages are created when your db upgrades from a db version lower than 85 to 85 or higher and the db was in version 85 before (85 is "downgradeable"). In this case, `onUpgrade` tries to add these columns (again) but they are already there.

This should ONLY occur in such an upgrade scenario, not when the db version stays the same. Can you confirm that?

SQLiteLog issues these messages because of my workaround for a "create column if not exists" which unfortunately SQLIte dies not support natively. Here is the code:

```Java
        /**
         * Helper for columns creation. This method ignores duplicate column errors
         * and is useful for migration situations
         */
        public void createColumnIfNotExists(final SQLiteDatabase db, final String table, final String columnDefinition) {
            try {
                db.execSQL("ALTER TABLE " + table + " ADD COLUMN " + columnDefinition);
                Log.i("[DB] Column '" + table + "'.'" + columnDefinition + "' created");
            } catch (SQLiteException sle) {
                if (!sle.getMessage().contains("duplicate column name")) {
                    throw sle;
                }
                Log.i("[DB] Column '" + table + "'.'" + columnDefinition + "' NOT created because it already exists");
            }
        }
```

the like with `db.execSQL` creates the log and I can do nothing to prevent that. WHen you log on info level or lower (like I do) you will see the log message at the bottom `([DB] Column ... NOT created because it already existsi`) directly after.

I dont know what I can do about that. ONly thing that comes to mind is to convert the "NOT created because it already exists" to a WARN level message (so it is also always present in the log when the other one is) and adjusting the log text a bit so the reader knows that although this is a warning message everything is all right. What do you think? 

However, I could

**Comment by eddiemuc** [comment]
A new idea came to my mind: we could introduce a new function `Log.iAlways` which, when called, issues a log on info level ALWAYS (regardless of debug and log settings). The above Log could use that function. Also I would then log some very basic startup information always using this function (e.g. the used database version or whether a db upgrade was performed).

What do you think?

**Comment by eddiemuc** [comment]
Hi @moving-bits I added PR #8998 for this. I also smuggled in other small improvements for logging (yes I am a bit obsessed currently with logging, this comes from working on #8632  :-)):
* I renamed the name of the file c:geo scans for log properties to `log-properties.txt`. Reason is that files with this name can be uploaded/shared in github (in contrast to files ending with `.properties`), and also the default handling of `.txt` files seems better on android than with `.properties` (opens text viewers instead of c:geo on click)
* I added a parameter to configure the maximum stack depth to add to log messages when "callerinfo" is active (I set the default to 4).
* I made a few startup log messages from `DBHelper` (which I consider essential) call the new `iForce` log method instead of `i` method

Hope you are ok with these changes?

**Comment by moving-bits** [comment]
Looks useful to me, thanks for implementing. Do we have `log.properties` on `release` branch or on `master` only? If already on `release`, then #8998 should be targeted to `release` as well IMHO.

**Comment by eddiemuc** [comment]
@moving-bits you are right, needs to be on release as well. Please don't merge yet, I have to look into this (whether this is compatible with release and such). I will set the PR to draft

**Comment by moving-bits** [comment]
> Hi @moving-bits , these messages are created when your db upgrades from a db version lower than 85 to 85 or higher and the db was in version 85 before (85 is "downgradeable"). In this case, `onUpgrade` tries to add these columns (again) but they are already there.
> 
> This should ONLY occur in such an upgrade scenario, not when the db version stays the same. Can you confirm that?

Thanks for cross-checking. Yeah, maybe I'm currently switching between `master` and `release` branches too frequently :-)
Haven't seen any more messages on further startups while still being on `master` branch.

**Comment by eddiemuc** [comment]
Hi @moving-bits , I created a new PR for release branch #9001 since not all changes in the other PR can be migrated to release (in release, there is not yet a method `createColumnIfNotExists`).
Once this PR is merged to release and back to master, I can then modify the original PR to use the new Log functions to issue the informative message after a possible "duplicate column warning".

Ok?

**Comment by moving-bits** [comment]
@eddiemuc 
Ok, is merged to `release` and `master`. I had to resolve some conflicts on master, so please cross-check if everything's as intended.

**Comment by eddiemuc** [comment]
@moving-bits master looks good, thanks for doing merge :-)
I changed PR #8998 accordingly, it can now be merged to master (after CI checks pass)
